llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020.ll

```diff
@@ -235,7 +235,7 @@ attributes #3 = { nounwind }
 ; CHECK: !sycl.specialization-constants = !{![[#ID0:]], ![[#ID1:]], ![[#ID2:]], ![[#ID3:]], ![[#ID_COMPOS3:]], ![[#ID4:]], ![[#ID5:]]
 ;
 ; CHECK-DEF: !sycl.specialization-constants-default-values = !{![[#ID4:]], ![[#ID5:]], ![[#ID6:]], ![[#ID7:]], ![[#ID_COMPOS3_DEFAULT:]], ![[#ID8:]], ![[#ID9:]]
-; CHECK-RT-NOT: !sycl.specialization-constants-default-values
+; CHECK-RT: !sycl.specialization-constants-default-values
```

Comment:
Ideally, we should be unifying `CHECK-DEF` and `CHECK-RT` for that metadata into a common `CHECK`, i.e. we should check content of the metadata in both modes.

However, this test is quite complicated and messy already, I plan to refactor those LIT tests to make them cleaner and simpler. Therefore, I'm ok to accept this test as-is.